components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeService.java

```diff
@@ -306,6 +311,288 @@ public Scope updateScope(Scope updatedScope) throws IdentityOAuth2ScopeException
         return updatedScope;
     }
 
+    /**
+     * Get OAuth scope consent given for an application by the user.
+     *
+     * @param userId        User Id.
+     * @param appId         Application Id.
+     * @param userTenantId  Tenant Id.
+     * @return  {@link OAuth2ScopeConsentResponse}.
+     * @throws IdentityOAuth2ScopeException
+     */
+    public OAuth2ScopeConsentResponse getUserConsentForApp(String userId, String appId, int userTenantId)
+            throws IdentityOAuth2ScopeException {
+
+        validateUserId(userId);
+        validateAppId(appId);
+        try {
+             UserApplicationScopeConsentDO userConsent = OAuthTokenPersistenceFactory.getInstance()
+                     .getOAuthUserConsentedScopesDAO()
+                    .getUserConsentForApplication(userId, appId, userTenantId);
+             OAuth2ScopeConsentResponse consentResponse = new OAuth2ScopeConsentResponse(userId, appId, userTenantId,
+                     userConsent.getApprovedScopes(), userConsent.getDeniedScopes());
+             if (log.isDebugEnabled()) {
+                 log.debug("Successfully retrieved the user consent for userId : " + userId + " and appId: "
+                         + appId);
+             }
+             return consentResponse;
+        } catch (IdentityOAuth2ScopeConsentException e) {
```

Comment:
handle client exceptions from lower levels

Reply:
Fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1548/commits/27ddbc8fed9c4368ffe7a826b857537a6b8ffff0